modules/proposals/codex/src/lib.rs

```diff
@@ -24,9 +24,10 @@ use rstd::marker::PhantomData;
 use rstd::prelude::*;
 use rstd::vec::Vec;
 use srml_support::{decl_error, decl_module, decl_storage, ensure};
+use system::RawOrigin;
 
 /// 'Proposals codex' substrate module Trait
-pub trait Trait: system::Trait + proposal_engine::Trait {}
+pub trait Trait: system::Trait + proposal_engine::Trait + proposal_discussion::Trait {}
```

Comment:
Why is `proposal_discussion::Trait` constraint needed here? I think we do need a trait for the discussion module interface itself, such that calls are mockable, but not sure why module trait is needed.

Reply:
I didn't plan to use mocks for the discussion modules. It is a small subsystem designed to be used within proposal modules. And module `Trait` is a Parity default way to use modules.

Reply:
Not relevant anymore, because we need to inherit from the `discussion::Trait` because we will store **discussions** to **proposals** map in the **codex** module.